### PR TITLE
Fix brokenness introduced by merging attributes as normal attributes

### PR DESCRIPTION
**Pull request by vuntz**
This is required for proper stoney->tex upgrade.

### CODE REVIEW

**Comment by vuntz** [comment]
Marking as WIP as I'm getting a different failure caused by this patch :/

**Comment by vuntz** [comment]
Found the issue. Seems to work now.

File: chef/cookbooks/database/recipes/crowbar.rb [hunk 1]
```diff
@@ -17,7 +17,35 @@
...
+    merge_onto
+
+    # In all other cases, replace merge_onto with merge_with
+  else
+    merge_with
+  end
+end
+
+if Chef::VERSION.split('.')[0].to_i >= 11
+  raise "Your chef version has hash_only_merge; consider removing the local copy."
```
Comment (rhafer) [inline chef/cookbooks/database/recipes/crowbar.rb]: I know this doesn't matter much currently but do you really want this recipe to always fail with chef 11?
Why not just log a warning and call the builtin hash_only_merge?

Note, I am fine with either way. I am just interested in the reasoning.

Comment (vuntz) [inline chef/cookbooks/database/recipes/crowbar.rb]: I just want us to remove the copy when we switch to chef 11, so we can clean this up :-) (ie, we'll stop removing chef 10)

Comment (vuntz) [inline chef/cookbooks/database/recipes/crowbar.rb]: I mean, stop supporting chef 10

**Comment by rhafer** [comment]
+1

**Pull request merged**